File: CP(Competetive Programming)/1.Practice/0.vscode/leetcode/1. TwoSum/TwoSum.java

class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        for(int i = 0; i<nums.length; i++){
            for(int j = i+1; j<nums.length; j++){
                if(nums[i]+nums[j]==target){
                   return new int[]{i,j};
                }
            }
        }
        return nums;
    }

    public static void main(String[] args){
        int[] ar = {3, 2, 4};
        int target = 6;
        TwoSum obj = new TwoSum();
        int[] a = obj.twoSum(ar, target);
        for(int i = 0; i<a.length; i++){
            System.out.println(a[i]);
        }
        //System.out.println();
    }
}

File: CP(Competetive Programming)/1.Practice/0. vscode/leetcode/1. TwoSum/TwoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        value_index = {}
        # nums.sort()
        for index in range(len(nums)):
            value_index[nums[index]] = index
        # for key, val in value_index.items():
        #     print(key, " ", val)
        for index in range(len(nums)):
            if target - nums[index] in nums and index < nums.index(target - nums[index]):
                ans.append(index)
                ans.append(nums.index(target - nums[index]))
            elif nums.count(nums[index]) > 1 and not ans and target - nums[index] in nums:
                ans.append(index)
                ans.append(value_index[nums[index]])
        return ans


arr = [3, 2, 3]
target = 6
obj = Solution()
ans = obj.twoSum(arr, target)
print(ans)
